src/app/doctors/views/table-view/table-view.component.ts

```diff
@@ -9,7 +12,18 @@ import { TABLE_HEADERS } from 'src/app/shared/constants';
 })
 export class TableViewComponent {
   @Input() doctors!: Doctor[];
-  isAdmin: boolean = false;
+  isAdmin!: boolean;
+
+  constructor(
+    private authService: AuthService,
+    private firebaseService: FirebaseService
+  ) {
+    this.authService.isAdmin.subscribe((data) => (this.isAdmin = data));
```

Comment:
### Issue
While subscribing to the `isAdmin` Observable is a valid approach, you don't handle the unsubscription.
It would be better to leverage the `async` pipe in you component template.

### Suggested change:
- Component template
```html
      <td *ngIf="isAdmin$ | async">
        <input
          type="checkbox"
          [value]="doctor.top"
          [checked]="doctor.top"
          (change)="onTopDoctorChange(doctor)"
        />
      </td>
````
- Component
```typescript
export class TableViewComponent {
  @Input() doctors!: Doctor[];
  isAdmin$!: Observable<boolean>;

  constructor(
    private authService: AuthService,
    private firebaseService: FirebaseService
  ) {
   this.isAdmin$ = this.authService.isAdmin;
    }